x-pack/platform/plugins/shared/alerting/server/alerts_client/lib/build_ongoing_alert.ts

```diff
@@ -81,6 +85,9 @@ export const buildOngoingAlert = <
   ActionGroupIds,
   RecoveryActionGroupId
 >): Alert & AlertData => {
+  // Sets array fields to empty arrays if previously reported in the existing alert
+  // but not present in the payload
+  replaceEmptyAlertFields(alert, payload);
```

Comment:
I wonder if we could do this as part of the `deepmerge` we do at the end? There's an `arrayMerge` option that we could customize https://www.npmjs.com/package/deepmerge

Reply:
I tried to update this in `deepmerge` but it causes some issue bc it clears out arrays from data we need like spaceId and tags, and I'm not able to differentiate between the objects we are passing into the deepmerge. Maybe I am missing something tho? If you an idea of how you want to do this in deepmerge